src/main/java/ai/promoted/delivery/client/PromotedDeliveryClient.java

```diff
@@ -133,57 +134,110 @@ private PromotedDeliveryClient(String deliveryEndpoint, String deliveryApiKey,
   }
 
   /**
-   * Used to call Delivery API. Takes the given list of Content and ranks it.
+   * Used to call Delivery API. Ranks the given list of Content.
    *
    * @param deliveryRequest the delivery request
    * @throws DeliveryException when any exception occurs
    */
   public DeliveryResponse deliver(DeliveryRequest deliveryRequest) throws DeliveryException {
+    DeliveryPlan plan = plan(deliveryRequest.isOnlyLog(), deliveryRequest.getExperiment());
+    prepareRequest(deliveryRequest, plan);
 
-    boolean isShadowTraffic = shouldSendShadowTraffic();
-    
+    Response apiResponse = null;
+    if (plan.useApiResponse()) {
+      try {
+        apiResponse = callDeliveryAPI(deliveryRequest);
+      } catch (DeliveryException ex) {
+        LOGGER.log(Level.WARNING, "Error calling Delivery API, falling back", ex);
+      }
+    }
+    return handleSdkAndLog(deliveryRequest, plan, apiResponse);
+  }
+
+  /**
+   * Returns a {@code DeliveryPlan} that determines SDK execution for
+   * {@link #deliver(DeliveryRequest) }.
+   *
+   * <p>This method provides support for reusing/overriding parts of
+   * {@link #deliver(DeliveryRequest) }.  Most users should use the {@code deliver} method instead.
+   * </p>
+   *
+   * @param onlyLog if true, the SDK Response will use the SDK-side paged response
+   * @see #deliver
+   */
+  public DeliveryPlan plan(boolean onlyLog, @Nullable CohortMembership experiment) {
+    boolean useApiResponse = !onlyLog && shouldApplyTreatment(experiment);
+    // TODO - why does checkCohortMembership copy?
```

Comment:
Isn't that function entirely deleted now?

Reply:
Yea.  I moved renamed it.  I don't know why it's needed.